fastapi/dependencies.py

```diff
@@ -19,8 +19,20 @@
     from .models.fastapi_endpoint import FastapiEndpoint
 
 
-def odoo_env() -> Environment:
-    yield odoo_env_ctx.get()
+def company_id() -> int | None:
+    """This method may be overriden by the FastAPI app to make the company
+    available for your endpoint. By default, the company defined on the
+    endpoint record is used.
+    """
+    return None
+
+
+def odoo_env(company_id: Annotated[int | None, Depends(company_id)]) -> Environment:
+    env = odoo_env_ctx.get()
+    if company_id:
```

Comment:
```suggestion
    if company_id is not None:
```

Pas grave dans ce cas mais une bonne habitude à prendre (pour les situations où on veut distinguer 0 et None).